app/src/main/java/com/example/calculator/Calculator.kt

```diff
@@ -0,0 +1,208 @@
+                      listOfNestedStrings: MutableList<String>,
+                      line: String?) {
+    for (nest in listOfNestedStrings) {
+        if (nest.contains("nested")) {
+            var result = proceedOrdinaryOperations(calculator,
+                parseNest(nest,listOfNestedStrings)
+                    .replace("(","")
+                    .replace(")","")
+                    .replace("|", ""))
+            when (result) {
+                is CommandResult.Error -> {
+                    println(result.message)
+                    return
+                }
+                is CommandResult.Success -> {
+                    val value = calculator.getResult()
+                    listOfNestedStrings[listOfNestedStrings.indexOf(nest)] = value.toString()
+                }
+            }
+        } else {
+            var result = proceedOrdinaryOperations(calculator,
+                nest.replace("(","")
+                    .replace(")",""))
+            when (result) {
+                is CommandResult.Error -> {
+                    println(result.message)
+                    return
+                }
+                is CommandResult.Success -> {
+                    val value = calculator.getResult()
+                    listOfNestedStrings[listOfNestedStrings.indexOf(nest)] = value.toString()
+                }
+            }
+        }
+    }
+    if (line!!.contains("nested")) {
+        var lastLine = line.substring(0,line.indexOf("n")) +
+                listOfNestedStrings.last().toString() +
+                line.substring(line.lastIndexOf("|") + 1)
+        calculator.resetResult()
+        if (lastLine.first().toString() == "-") {
+            lastLine = "0$lastLine"
+        }
+        val result = proceedOrdinaryOperations(calculator,lastLine)
+        when (result) {
+            is CommandResult.Error -> {
+                println(result.message)
+                return
+            }
+            is CommandResult.Success -> {
+                val value = calculator.getResult()
+                listOfNestedStrings.drop(listOfNestedStrings.count())
+                println(calculator.getResult())
+            }
+        }
+    }
+}
+
+fun parseNest(nest: String, listOfNestedStrings: MutableList<String>): String {
+    val indexOfNeededNest = nest
+        .substring(nest.indexOf("|") + 1,
+            nest
+                .lastIndexOf("|"))
+        .toInt()
+    return nest.substring(0,nest.indexOf("n")) +
+            listOfNestedStrings[indexOfNeededNest] +
+            nest.substring(nest.lastIndexOf("|") + 1)
+}
+
+fun proceedOrdinaryOperations(calculator: CommandProcessor, line: String?): CommandResult {
+    val operationsList = Validator().validateOperationList(line)
+    val numbersList = Validator().validateNumbersList(line)
+    calculator.setIsFirstDigit(Validator().checkFirstSymbol(line))
+    return calculator.startProcessing(
+        numbersList,
+        operationsList ?: emptyList()
+    )
```

Comment:
3 раза создается валидатор
```suggestion
    val validator = Validator()
    calculator.setIsFirstDigit(validator.checkFirstSymbol(line))
    return calculator.startProcessing(
        validator.validateNumbersList(line),
        validator.validateOperationList(line) ?: emptyList()
    )
```

или даже так
```suggestion
     return with(Validator()) {
        calculator.setIsFirstDigit(checkFirstSymbol(line))
        calculator.startProcessing(
            validateNumbersList(line),
            validateOperationList(line) ?: emptyList()
        )
    }
```